test/metabase/query_processor/middleware/parameters_test.clj

```diff
@@ -23,7 +23,28 @@
           :native          {:query "WOW", :parameters ["My Param"]}
           :user-parameters ["My Param"]}
          (#'parameters/move-top-level-params-to-inner-query
-          {:type :native, :native {:query "WOW"}, :parameters ["My Param"]}))))
+          {:type :native, :native {:query "WOW"}, :parameters ["My Param"]})))
+  (testing "when top-level query is a model"
```

Comment:
Would it make sense to test also with native query?

Reply:
I tweaked this PR; native queries don't work before or after this change. I've filed #40011 to fix this for native models eventually.